web/basics/binding_to_a_complex_object/my_element.dart

```diff
@@ -2,32 +2,40 @@
 library web.basics.binding_to_a_complex_object;
 
 import 'package:polymer/polymer.dart';
+import 'package:web_components/web_components.dart' show HtmlImport;
 
-class Name {
+class Name extends JsProxy {
   String first, last;
   Name(this.first, this.last);
 }
 
-class Company {
+class Company extends JsProxy {
   String name;
   Company(this.name);
 }
 
-class Person {
+class Person extends JsProxy {
   Name name;
   String title;
   Company company;
   Person(this.name, this.title, this.company);
 }
 
-@CustomTag('my-element')
+class Foo extends JsProxy {
+  String bar;
+  Foo(this.bar);
+}
+
+@PolymerRegister('my-element')
 class MyElement extends PolymerElement {
-  @observable Person person;
+  @property
+  Person person;
   MyElement.created() : super.created() {
-    person = new Person(
-        new Name('Kathy','Walrath'),
-        'Writer Extraordinaire',
-        new Company('Google')
-    );
+    person = new Person(new Name('Kathy', 'Walrath'), 'Writer Extraordinaire',
+        new Company('Google'));
+    this.notifyPath('person.name.first', person.name.first);
```

Comment:
To clarify, `notifyPath('person', person);` works in this scenario because the whole person did change. If you only changed a property of the person you would need to call `notifyPath` with the exact path to the property that changed. 

For maps the first option always works as a side-effect of how the maps are passed to JS (a new JsObject is created each time). This will hopefully not always be the case though.